src/libraries/Native/Unix/System.Native/pal_io.c

```diff
@@ -263,6 +263,8 @@ static int32_t ConvertOpenFlags(int32_t flags)
         ret |= O_TRUNC;
     if (flags & PAL_O_SYNC)
         ret |= O_SYNC;
+    if (flags & PAL_O_DIRECT)
+        ret |= O_DIRECT;
```

Comment:
Does this exist on macOS?

Reply:
The man page states:
```
O_DIRECT I/Os should never be run concurrently with the fork(2)
       system call, if the memory buffer is a private mapping (i.e., any
       mapping created with the mmap(2) MAP_PRIVATE flag; this includes
       memory allocated on the heap and statically allocated buffers).
       Any such I/Os, whether submitted via an asynchronous I/O
       interface or from another thread in the process, should be
       completed before fork(2) is called.  Failure to do so can result
       in data corruption and undefined behavior in parent and child
       processes.
```
How are you guaranteeing this, in our tests but also for use in the wild?  `fork` is used by Process.Start.